react-blog/src/store/reducers/user.ts

```diff
@@ -12,16 +14,18 @@ const initialState: IUser = {
   email: null,
 };
 
-const userReducer = (state = initialState, action: AnyAction) => {
-  switch (action.type) {
-    case "SET_USER":
-      return { ...state, ...action.payload };
-    case "RESTORE_USER":
-      return initialState;
-
-    default:
+export const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {
+    setUser: (state, action) => {
+      return {...state, ...action.payload};
+  },
+    restoreUser: (state) => {
       return state;
-  }
-};
+    },
```

Comment:
return initialState for restoring user as it contains our default info
In other case this method do nothing.